packages/core/src/dispatcher.ts

```diff
@@ -306,8 +327,12 @@ export class Dispatcher {
 
     // Lookup CID in cache before looking it up IPFS
     const resolutionPath = `${asCid}${path || ''}`
-    const cachedDagNode = this.dagNodeCache.get(resolutionPath)
-    if (cachedDagNode) return cloneDeep(cachedDagNode)
+    const fromCache = this.ipldCache.getUnresolved(resolutionPath)
+    if (fromCache) {
+      Metrics.count(IPFS_CACHE_HIT, 1)
+      return cloneDeep(fromCache.record)
```

Comment:
do you think we really need to clone this here?  How likely is it that any callers are modifying the record?

Reply:
Haha, I had the same question to myself. I thought you would be against removing it :)

Reply:
Ideally the type system would let us specify the return value as immutable, but alas

Reply:
Just sprinkle some rust on it 😆